dataset/dataset.go

```diff
@@ -216,6 +216,29 @@ func (c *Client) GetDatasets(ctx context.Context, userAuthToken, serviceAuthToke
 	return
 }
 
+// PutDataset update the dataset
+func (c *Client) PutDataset(ctx context.Context, userAuthToken, serviceAuthToken, collectionID, datasetID string, d DatasetDetails) error {
+	uri := fmt.Sprintf("%s/datasets/%s", c.hcCli.URL, datasetID)
+
+	clientlog.Do(ctx, "updating dataset", service, uri)
+
+	payload, err := json.Marshal(d)
+	if err != nil {
+		return errors.Wrap(err, "error while attempting to marshall dataset")
+	}
+
+	resp, err := c.doPutWithAuthHeaders(ctx, userAuthToken, serviceAuthToken, collectionID, uri, payload)
+	if err != nil {
+		return errors.Wrap(err, "http client returned error while attempting to make request")
+	}
+	defer closeResponseBody(ctx, resp)
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.Errorf("incorrect http status, expected: 200, actual: %d, uri: %s", resp.StatusCode, uri)
```

Comment:
other methods in dataset api client return a DatasetAPIResonse, I think it would be worth returning it here too, for consistency reasons:
```
 return NewDatasetAPIResponse(resp, uri)
```